menu/menu_displaylist.c

```diff
@@ -3621,8 +3621,19 @@ static int menu_displaylist_parse_cores(
       return 0;
    }
 
-   str_list = dir_list_new(info->path, info->exts,
-         true, settings->bools.show_hidden_files, true, false);
+   if (string_is_empty(settings->paths.directory_libretro_distro))
+      str_list = dir_list_new(info->path, info->exts, 
+                  true, settings->bools.show_hidden_files, true, false);
+   else
+   {
+      struct string_list *list1 = dir_list_new(info->path, info->exts, 
+                                    true, settings->bools.show_hidden_files, true, false);
+      struct string_list *list2 = dir_list_new(settings->paths.directory_libretro_distro, info->exts, 
+                                    true, settings->bools.show_hidden_files, true, false);
+      str_list = dir_list_concatenate(list1, list2);
+      string_list_free(list1);
+      string_list_free(list2);
+   }
 
```

Comment:
basically this is all the code needed to concatenate two paths so it shows as one